openmc_plotter/plotmodel.py

```diff
@@ -735,26 +847,12 @@ class PlotView(openmc.lib.plot._PlotBase):
         Indicates whether or not tallies are displayed as contours
     tallyContourLevels : str
         Number of contours levels or explicit level values
-    selectedTally : str
-        Label of the currently selected tally
     """
 
-    def __init__(self, origin, width, height):
-        """ Initialize PlotView attributes """
-
-        super().__init__()
-
-        # View Parameters
-        self.level = -1
-        self.origin = origin
-        self.width = width
-        self.height = height
-        self.h_res = 1000
-        self.v_res = 1000
-        self.aspectLock = True
-        self.basis = 'xy'
-
+    def __init__(self):
+        """Initialize PlotViewIndependent attributes"""
         # Geometry Plot
+        self.aspectLock = True
         self.colorby = 'material'
         self.masking = True
```

Comment:
@pshriwise do you know why this was initially set to True? And is that the desired behavior? I ask because the way that this code is currently setup, by setting this to true it actually runs `applyChanges()` during creation and stores the this initial view as a previous view upon opening. This means that the "undo" button in the menu already has 1 undo possible without really having a view to undo.

Reply:
Hrmm yeah that's not ideal and potentially confusing. 

As for why this particular line is here, I think it's simply being set with an initial value like the other attributes. It's been a while, so I'm not sure why setting this particular value (as opposed to any of the others) would trigger a callback to `applyChanges()`. Can you elaborate how setting this to `True` triggers the `applyChanges()` call?

Reply:
Setting it to `True` essentially "turns it on" as if it were a change in the toggle. The relevant lines are where it's added to the menu [here](https://github.com/kkiesling/plotter/blob/separate_model_view/openmc_plotter/main_window.py#L327) and then the actual updating of the setting [here](https://github.com/kkiesling/plotter/blob/separate_model_view/openmc_plotter/main_window.py#L738). I don't fully understand why it behaves like this to be honest, other than it is caused by this being initialized to `True`. The other 3 settings in the same part of the menu that are setup the same way are initialized to `False` and do not trigger a change to `applyChanges()`.

Reply:
Got it. Thanks for the links! They helped me track down the issue.

I discovered that the flow which causes this is:

1. [Model is loaded when initializing `MainWindow`](https://github.com/kkiesling/plotter/blob/9ae7a7620b14625f1537b9dfc130b09097b02061/openmc_plotter/main_window.py#L58)
2. [`PlotIndependent.masking` is set to `True` for the initial current and active view.](https://github.com/kkiesling/plotter/blob/9ae7a7620b14625f1537b9dfc130b09097b02061/openmc_plotter/plotmodel.py#L862)
3. [The GUI continues loading in `MainWindow.loadGui` and `MainWindow.updateEditMenu` is called](https://github.com/kkiesling/plotter/blob/9ae7a7620b14625f1537b9dfc130b09097b02061/openmc_plotter/main_window.py#L99)
4. (The crux) `MainWindow.updateEditMenu` calls the `setChecked` method of the masking check box to ensure it's state matches that of the view. If this call changes the state of that check box, then the callback method is going to trigger an `applyChanges` call and create a new view.

It seems to me that the default value of the masking check box in the color dialog doesn't match the view setting and results in the creation of a new view.

I verified this by adding this line in `MainWindow.createMenuBar`

```python
self.maskingAction.setChecked(True)
```

before the item is connected to its callback method. This removed the superfluous previous view for me.

I'd say that fix is sufficient for now, but in the future I think it would be nice to have some internal setting that prevents any updates while the state of these data structures is initialized would be good.

Reply:
It does indeed look like this is pre-existing behavior. I'll make an issue so we can clean it up in a subsequent PR.